src/httpc.c

```diff
@@ -173,9 +173,10 @@ httpc_request_delete(struct httpc_request *req)
 int
 httpc_set_header(struct httpc_request *req, const char *fmt, ...)
 {
+	static __thread char header[TT_STATIC_BUF_LEN*8];
 	va_list ap;
 	va_start(ap, fmt);
-	const char *header = tt_vsprintf(fmt, ap);
+	vsnprintf(header, TT_STATIC_BUF_LEN*8, fmt, ap);
```

Comment:
vsnprintf() will return a number more then 8191 (because doesn't count '\0') if the output was truncated. It worth to check it here and set a diag (`diag_set(IllegalParams, <…>)`) in the case and return -1.

Reply:
Change to HTTP_HEADER_LEN